### PR TITLE
[TorchToTMTensor] Accept 1/sqrt(headDim) scale for SDPA

### PR DESCRIPTION
**Pull request by jtuyls**
This PR improves the scaled_dot_product_attention conversion in TorchToTMTensor to accept explicit scale values that match PyTorch's default 1/sqrt(headDim), not just scale=None, see https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html.

Breaking change: Support for scale=1.0 is removed. This was likely incorrect as it's not the default as was mentioned — also, for example, IREE's downstream AttentionOpConversion always applies 1/sqrt(headDim) scaling internally, so accepting scale=1.0 would silently produce wrong results (the output would be scaled by 1/sqrt(headDim) when the user expected no scaling): https://github.com/iree-org/iree/blob/f4b596d5b2a8813948f6dab18a36403e768eb5e1/compiler/plugins/input/Torch/InputConversion/ConvertTMTensorToLinalgExt.cpp#L174.

Note that this change accepts scale values that match 1/sqrt(headDim) within a relative tolerance of 1e-6 (~10x float32 machine epsilon), which handles:
- Different computation orders producing slightly different values
- Float64 → float32 → float64 round-trips through serialization

### CODE REVIEW

**Comment by jtuyls** [comment]
Could someone help review? @vivekkhandelwal1 @zjgarvey ?

**Comment by zjgarvey** [comment]
This seems fine to me as long as we don't see a dynamic head dim in practice (maybe a runtime check would be better if we need to support this case). 


@Groverkss , I'd like to make sure you are okay with this change before landing.

**Comment by jtuyls** [comment]
> This seems fine to me as long as we don't see a dynamic head dim in practice (maybe a runtime check would be better if we need to support this case).

I would expect sdpa with dynamic head dim (if seen at all, I don't see how that would even work) to have scale parameter set to None as the scale then likely needs to be computed dynamically as well. 

NOTE: We could also just enable any scale here, however I didn't see an example with that and it wasn't enabled here so there might be issues that I am not aware of (I know fore sure IREE doesn't handle it and always uses 1/sqrt(headDim)) so I didn't go for that for now.

**Comment by jtuyls** [comment]
@Groverkss Could you have a look?

**Comment by Groverkss** [comment]
Can we just accept scale as a parameter and if it's not passed use 1/sqrt(d)?  Probably an oversight in the past.

**Comment by jtuyls** [comment]
> Can we just accept scale as a parameter and if it's not passed use 1/sqrt(d)? Probably an oversight in the past.

Yeah, but then I would have to change the op and plumb that through and I can do that as a follow up or create an issue for it (I am not in a hurry for that as my model just uses 1/sqrt(headDim), so that's all I need for now, might look at it when I have time). I think it shouldn't block this PR as this just fixes an issue (default_scale is not 1.0, but 1/sqrt(headDim)).

**Comment by Groverkss** [comment]
Sure that's okay then, feel free to land it
